Composer/packages/tools/language-servers/language-generation/src/LGServer.ts

```diff
@@ -443,13 +444,12 @@ export class LGServer {
     const range = getRangeAtPosition(document, position);
     const wordAtCurRange = document.getText(range);
     const endWithDot = wordAtCurRange.endsWith('.');
-    const lgFile = this.getLGDocument(document)?.index();
+    const lgFile = LGParser.parseText(document.getText(), '', this.getImportResolver(document));
     if (!lgFile) {
       return Promise.resolve(null);
     }
-    const { templates } = lgFile;
 
-    const completionTemplateList: CompletionItem[] = templates.map(template => {
+    const completionTemplateList: CompletionItem[] = lgFile.allTemplates.map(template => {
```

Comment:
we can extend return values of `this.getLGDocument(document)?.index()` to cover allTemplates
```
{
  diagnostics,
  templates,      // file itself's
  allTemplates,  // include imports
}
```